### PR TITLE
Limit TaskHeaders sent

### PR DESCRIPTION
**Pull request by Elfoniok**
Signed-off-by: Lukasz Foniok <lukaszfoniok@gmail.com>

### CODE REVIEW

File: golem/network/p2p/peersession.py [hunk 1]
```diff
@@ -438,7 +439,16 @@ def _react_to_peers(self, msg):
...
     def _react_to_get_tasks(self, msg):
         tasks = self.p2p_service.get_tasks_headers()
-        self.send(message.MessageTasks(tasks))
+        max_ind = len(tasks) - 1
+        if max_ind < 0:
+            return
+        tasks_to_send = []
+        if len(tasks) > TASK_HEADERS_LIMIT:
+            for i in range(0, TASK_HEADERS_LIMIT):
+                tasks_to_send.append(tasks[random.randint(0, max_ind)])
```
Comment (ggruszczynski) [inline golem/network/p2p/peersession.py]: random.randint(0, max_ind) can return the same number several times.
Please, use drawing without repetition.

**Review by ggruszczynski — CHANGES_REQUESTED** [review]
Please:
1) use drawing without repetition.
2) to keep a good test coverage add a simple unit test 
Have a look at test_peerseession and test_taskserver.test_add_task_header --> get_example_task_header.

File: golem/network/p2p/peersession.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/network/p2p/peersession.py]: In Python 3.6, there is a `random.choices(collection)` function. We can backport it to 3.5 from [here](https://github.com/python/cpython/blob/master/Lib/random.py#L342)

Comment (Elfoniok) [reply golem/network/p2p/peersession.py]: But choices is with repetition, no? I want to try random.sample is this all right?
https://docs.python.org/3/library/random.html#random.sample

File: tests/golem/network/p2p/test_peersession.py [hunk 2]
```diff
@@ -186,6 +187,30 @@ def test_interpret(self):
...
+        peer_session._react_to_get_tasks(Mock())
+        assert not peer_session.send.called
+
+        peer_session.p2p_service.get_tasks_headers.return_value = list(
+            range(0, 100))
+        peer_session._react_to_get_tasks(Mock())
+
+        sent_tasks = peer_session.send.call_args_list[0][0][0].tasks_array
+        assert len(sent_tasks) <= TASK_HEADERS_LIMIT
+        assert len(sent_tasks) == len(set(sent_tasks))
```
Comment (ggruszczynski) [inline tests/golem/network/p2p/test_peersession.py]: what is the purpose of  `assert len(sent_tasks) == len(set(sent_tasks))` ?
I think that `assert len(sent_tasks) == TASK_HEADERS_LIMIT` should be enought

Comment (Elfoniok) [reply tests/golem/network/p2p/test_peersession.py]: I would but it's better to check if all entries are unique, as you pointed out without repetition.

**Review by ggruszczynski — APPROVED** [review]

File: golem/network/p2p/peersession.py [hunk 3]
```diff
@@ -441,7 +442,14 @@ def _react_to_peers(self, msg):
 
     def _react_to_get_tasks(self, msg):
         tasks = self.p2p_service.get_tasks_headers()
-        self.send(message.MessageTasks(tasks))
+        max_ind = len(tasks) - 1
+        if max_ind < 0:
```
Comment (jiivan) [inline golem/network/p2p/peersession.py]: Maybe this would be more readable?
```python
if not tasks:
    return
```

**Review by jiivan — APPROVED** [review]
Small comment, not a stopper.

**Comment by maaktweluit** [comment]
@Elfoniok This PR looks ready for merge, anything left to fix? If so, can i help?

**Pull request merged**